### ISSUE TITLE
[feat] In dark mode, the tauri app will flash white at the moment it starts, hope some change

### ISSUE THREAD

**Issue by kingwingfly** [tauri-apps/tauri#6027]
### Describe the problem

Poor in English.  
When the system is in dark mode, it is naturally that the background of the app is nearly black.  
However, at the moment the tauri app starts, it will slightly shine brightly for some milliseconds.  

If there is some method to avoid this? Actually I'm quite new in vue. 

### Describe the solution you'd like

On daytime, keep.  
On night(dark mode), avoid the shining at the moment it starts.

### Alternatives considered

List some tips for solving this little strange problem to a easy-gotten place.

### Additional context

Maybe there's a way to solving it that may be just very basic for a web developer, if so, I'm so sorry for wasting the time but it would be so kind that someone could teach me!

**Comment by amrbashir** [comment]
see https://github.com/tauri-apps/tauri/issues/1564

**Comment by dionysuzx** [comment]
I solved this issue in my Svelte 5 + Tauri 2.0 app like so:

1. Set the `"core:window:allow-show"` permission under `capabilities/default.json` like so:
    ```
    "permissions": ["core:default", "core:window:allow-show", "shell:allow-open"]
    ```

2. Show window on mount (using state rune / store):

    Homepage:
    ```
    <script lang="ts">
      import { onMount } from "svelte";
      import { window } from "../stores/window.svelte.js";
    
      onMount(() => {
        window.show();
      });
    </script>
    ```

    Writable store:
    ```
    import { getCurrentWebviewWindow } from "@tauri-apps/api/webviewWindow";
    
    let shown = $state(false);
    
    export const window = {
      get shown() {
        return shown;
      },
      async show() {
        if (!shown) {
          console.log("showing window");
          await getCurrentWebviewWindow().show();
          shown = true;
        } else {
          console.log("window already shown");
        }
      },
    };
    ```

Hope it helps, cheers! 🍺
